src/DynamoCoreWpf/Views/GuidedTour/PopupWindow.xaml.cs

```diff
@@ -40,6 +49,32 @@ public PopupWindow(PopupWindowViewModel viewModel, HostControlInfo hInfo)
             ContentRichTextBox.Width = popupViewModel.Width - (popupBordersOffSet * 4);
             HorizontalOffset = hInfo.HorizontalPopupOffSet;
             VerticalOffset = hInfo.VerticalPopupOffSet;
+
+            if (!string.IsNullOrEmpty(hInfo.HtmlPage))
+            {
+                ContentRichTextBox.Visibility = Visibility.Hidden;
+                Opened += PopupWindow_Opened;
+            }
+
+            Closed += PopupWindow_Closed;
+        }
+
+        ~PopupWindow()
```

Comment:
I don't think this will do what you want. Finalizers should be used for cleaning up unmanaged resources - the garbage collector will never invoke this finalizer because the events may still be keeping managed references - you should cleanup the events sometime in the managed lifetime, like when the popup closes or when you call dispose or unsubscribe or something.

Reply:
Thanks!